File: Lab4/4.py
# Check https://thecode.media/comb-sort/ firstly
def comb_sort(arr):
    gap = len(arr)
    shrink = 1.247
    sorted = False
    while gap >= 1 or not sorted:
        gap = int(gap / shrink)
        for i in range(len(arr) - gap):
            if arr[i] > arr[i + gap]:
                arr[i], arr[i + gap] = arr[i + gap], arr[i]
                sorted = False
        if gap >= 1:
            sorted = True
    return arr


arr = [10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
print(comb_sort(arr))

File: Semester_1/Labs/Lab4/4.cpp
#include <iostream>

void combSort(int a[])
{
    float shrink = 1.247;
    int length = sizeof(a) / sizeof(a[0]);
    float gapFactor = length / shrink;
    int gap;
 
    while (gapFactor > 1) {
        gap = (int)gapFactor;
        for (int i = 0; i < length - gap; i++) {
            if (a[i] > a[i + gap]) {
                std::swap(a[i], a[i + gap]);
            }
        }
        gapFactor /= shrink;
    }
}